app/plugin_install.go

```diff
@@ -140,8 +140,10 @@ func (a *App) removePlugin(id string) *model.AppError {
 
 	// Remove bundle from the file store.
 	storePluginFileName := filepath.Join("./plugins", manifest.Id) + ".tar.gz"
-	if err := a.RemoveFile(storePluginFileName); err != nil {
-		return model.NewAppError("removePlugin", "app.plugin.remove_bundle.app_error", nil, err.Error(), http.StatusInternalServerError)
+	if bundleExist, _ := a.FileExists(storePluginFileName); bundleExist {
```

Comment:
Wondering if we should test the `FileExists` error? If the file store wasn't available for some reason, I think it's fine to fail the removal, since it's possible it won't be "complete" and the plugin would just get reinstalled the next time.

In general, we can rely on the file store to be reliable, so having it fail spuriously for plugins that were never in the store is probably ok.

Reply:
At this point in the `removePlugin` the removal of the plugin is not reversible so it can't fail the complete process. It can return the error that would tell the client that the bundle is not in the file store or we could silently log it as a warning?

Given that for those that already had plugins installed before this new feature and go to remove it, it will always fail this check (which they won't understand why) and it will be a rare case that the filestore fails, I would be more inclined to silently log the "Bundle does not exist in FileStore" error rather than returning it to the client. Thoughts? @lieut-data

Reply:
Fast forwarding to a future where the file store backing is fully online, if we fail to remove it from the file store (even though it's gone from disk) it will just "come back" on server restart. So while we can't (easily) recover from the partial removal, I'd advocate for letting the client know we failed to fully remove it.

Reply:
Yes, the error from removing the bundle is already handle in the second `if`:
```
if err := a.RemoveFile(storePluginFileName); err != nil {
			return model.NewAppError("removePlugin", "app.plugin.remove_bundle.app_error", nil, err.Error(), http.StatusInternalServerError)
```
The outer `if` only checks if it's in the file Store: 
```
if bundleExist, _ := a.FileExists(storePluginFileName); bundleExist {
```
that's the error that would only happen in the backward compatibility cases or very rare file store access issue. This one is the one I propose logging but not returning to the client.

Reply:
@lieut-data what are your final thoughts on this error case ?

Reply:
> Given that for those that already had plugins installed before this new feature and go to remove it, it will always fail this check (which they won't understand why) and it will be a rare case that the filestore fails, I would be more inclined to silently log the "Bundle does not exist in FileStore" error rather than returning it to the client. Thoughts? @lieut-data

Not sure I understand why `a.FileExists(storePluginFileName)` would return an error in the first case you described above. It will return an error if the there store is offline, _i think_ 🤔 

I may be wrong but i think what @lieut-data is referring to is to check for the `err` value

```
if bundleExist, err := a.FileExists(storePluginFileName)
if err != nil {
    return model.NewAppError("...") 
}
```

Reply:
At the time they are trying to remove the plugin, it would return an error if the plugin bundle does not exist in the file store, which would be the case where they had it installed previous to this new feature. 
I believe in this case, there is not much value returning this error message to the client and we could silently log as a warning.

Reply:
I thought `FileExists` would _not_ return an `*model.AppError` err but a value of `false` if a file didn't exist, is that not true?

Reply:
You are correct. I got confused with the second call of `RemoveFile`🤦‍♀️

Adding the check for `err` now